internal/backends/sqlite/metadata/registry.go

```diff
@@ -353,6 +374,94 @@ func (r *Registry) CollectionRename(ctx context.Context, dbName, oldCollectionNa
 	panic("not implemented")
 }
 
+// indexesCreate creates an index in the collection.
+//
+// It does not hold the lock.
+func (r *Registry) indexesCreate(ctx context.Context, db *fsql.DB, c *Collection, indexes []IndexInfo) error {
+	err := db.InTransaction(ctx, func(tx *fsql.Tx) error {
+		for _, index := range indexes {
+			// check if index already exists
+			q := `SELECT EXISTS (SELECT 1 FROM sqlite_master WHERE type= 'index' and tbl_name = ? and name = ?)`
```

Comment:
Can this be checked from `Settings` rather than querying `sqlite_master`? Answer might be no :stuck_out_tongue:

Reply:
In principle, we can. My idea was to use the query here as we use it in the other place in this func, but actually checking settings might be better.

Reply:
Not super important if it works.

Reply:
It is important not to query the database after `initCollections`. Note that all other queries are inserts/updates/deletes; there are no selects.